File: tomas-flutter/lib/models/tukang.dart
import '../utils/json_value.dart';

class Tukang {
  final int idTukang;
  final String nama;
  final String? kategori;
  final String? lokasi;
  final String? alamat;
  final String? bio;
  final String? noHp;
  final double? tarif;
  final bool statusAktif;
  final String? fotoUrl;
  final double rating;
  final double? latitude;
  final double? longitude;

  Tukang({
    required this.idTukang,
    required this.nama,
    this.kategori,
    this.lokasi,
    this.alamat,
    this.bio,
    this.noHp,
    this.tarif,
    required this.statusAktif,
    this.fotoUrl,
    this.rating = 4.7,
    this.latitude,
    this.longitude,
  });

  factory Tukang.fromJson(Map<String, dynamic> j) => Tukang(
    idTukang: jsonInt(j['id_tukang']),
    nama: jsonString(j['nama'], fallback: 'Tukang'),
    kategori: jsonStringOrNull(j['kategori']),
    lokasi: jsonStringOrNull(j['lokasi']),
    alamat: jsonStringOrNull(j['alamat']),
    bio: jsonStringOrNull(j['bio']),
    noHp: jsonStringOrNull(j['no_hp']),
    tarif: jsonDoubleOrNull(j['tarif']),
    statusAktif: jsonBool(j['status_aktif']),
    fotoUrl: jsonStringOrNull(j['foto_url']),
    rating: jsonDouble(j['rating'], fallback: 4.7),
    latitude: jsonDoubleOrNull(j['latitude']),
    longitude: jsonDoubleOrNull(j['longitude']),
  );
}

File: admin-panel/app/Models/Tukang.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Tukang extends Model
{
    protected $table = 'tukang';
    protected $primaryKey = 'id_tukang';
    public $timestamps = true;

    protected $hidden = [
        'password',
    ];

    protected $fillable = [
        'nama', 'status_aktif', 'lokasi', 'alamat', 'kategori', 'bio', 'foto', 'tarif',
        'username', 'password', 'no_hp', 'no_ktp', 'foto_ktp', 'foto_selfie',
        'status_verifikasi', 'latitude', 'longitude', 'deposit_balance', 'deposit_minimum',
    ];

    protected $casts = [
        'deposit_balance' => 'decimal:2',
        'deposit_minimum' => 'decimal:2',
        'status_aktif' => 'boolean',
    ];

    public function orders()
    {
        return $this->hasMany(Order::class, 'id_tukang', 'id_tukang');
    }

    public function reviews()
    {
        return $this->hasManyThrough(Review::class, Order::class, 'id_tukang', 'id_order', 'id_tukang', 'id_order');
    }

    public function favorit()
    {
        return $this->hasMany(Favorit::class, 'id_tukang', 'id_tukang');
    }

    public function portfolios()
    {
        return $this->hasMany(Portfolio::class, 'id_tukang', 'id_tukang');
    }

    public function badges()
    {
        return $this->hasMany(BadgeAward::class, 'target_id', 'id_tukang')
            ->where('target_type', 'tukang');
    }

    public function supportChats()
    {
        return $this->hasMany(SupportChat::class, 'id_tukang', 'id_tukang');
    }

    public function isFavoritedBy($userId)
    {
        return $this->favorit()->where('id_user', $userId)->exists();
    }
}
